### PR TITLE
Fix column offset of unary operator AST node

### LINKED ISSUE #3564: chaining other commands after `nix-store` causes "NameError: name 'nix' is not defined"
## xonfig

<details>

```
$ xonfig                                                                                
+------------------+-----------------+
| xonsh            | 0.9.17          |
| Python           | 3.7.7           |
| PLY              | 3.11            |
| have readline    | True            |
| prompt toolkit   | 3.0.4           |
| shell type       | prompt_toolkit2 |
| pygments         | 2.5.2           |
| on posix         | True            |
| on linux         | True            |
| distro           | unknown         |
| on darwin        | False           |
| on windows       | False           |
| on cygwin        | False           |
| on msys2         | False           |
| is superuser     | False           |
| default encoding | utf-8           |
| xonsh encoding   | utf-8           |
| encoding errors  | surrogateescape |
+------------------+-----------------+
```

</details>

## nix-info -m

<details>

 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.35, NixOS, 20.09pre223023.fce7562cf46 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
 - channels(root): `"nixos-20.09pre223023.fce7562cf46, nixos-unstable-20.09pre218613.ae6bdcc5358"`
 - channels(das-g): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

</details>

## Expected Behavior
If `nix-store` is in `$PATH`, it can be used as part of longer command lines, too, with other commands chained after it.

## Current Behavior

If the command line continues after `nix-store` and its arguments, xonsh won't run `nix-store` and instead reports `NameError: name 'nix' is not defined`.

### Traceback (if applicable)

<details>

```
xonsh: To log full traceback to a file set: $XONSH_TRACEBACK_LOGFILE = <filename>
Traceback (most recent call last):
  File "/nix/store/y22pwfk39i9r22ga10c3ifbp345gdl0p-xonsh-0.9.17/lib/python3.7/site-packages/xonsh/base_shell.py", line 362, in default
    run_compiled_code(code, self.ctx, None, "single")
  File "/nix/store/y22pwfk39i9r22ga10c3ifbp345gdl0p-xonsh-0.9.17/lib/python3.7/site-packages/xonsh/codecache.py", line 67, in run_compiled_code
    func(code, glb, loc)
  File "<xonsh-code>", line 1, in <module>
NameError: name 'nix' is not defined
```

</details>

## Steps to Reproduce

1. make sure `nix-store` is in `$PATH`
    ```xonsh
    nix-store --version
    ```
    You should get some output like
    ```
    nix-store (Nix) 2.3.4
    ```
2. call the following command line
    ```xonsh
    nix-store --version or true
    ```

### PR DESCRIPTION
**Pull request by laloch**
Fixes #3564

### CODE REVIEW

File: news/fix-unary-position.rst [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+* <news item>
+
+**Removed:**
+
+* <news item>
+
+**Fixed:**
+
+* Logical sobprocess operators now work after long arguments (e.g. --version).
```
Comment (das-g) [inline news/fix-unary-position.rst:19]:
```suggestion
* Logical subprocess operators now work after long arguments (e.g. `--version`).
```

Comment (laloch) [reply news/fix-unary-position.rst]: Thanks! New system - no proofreader :wink:

File: tests/test_aliases.py [hunk 2]
```diff
@@ -121,6 +121,18 @@ def test_recursive_callable_partial_none(xonsh_execer, xonsh_builtins):
...
+        "echo --version or echo 'there'",
+        "echo --version || echo 'there'",
+        "echo -+version and echo 'there'",
+        "echo -+version && echo 'there'",
+        "echo -+version or echo 'there'",
+        "echo -+version || echo 'there'",
+        "echo -~version and echo 'there'",
+        "echo -~version && echo 'there'",
+        "echo -~version or echo 'there'",
+        "echo -~version || echo 'there'",
```
Comment (das-g) [inline tests/test_aliases.py:135]: These tests look sensible, but they **do pass even without the fix** (i.e., without the change from 04ecb2661f82612a32dd5a105b48da4aa2116e35), so it seems they test something else than what was fixed here.

Comment (laloch) [reply tests/test_aliases.py:135]: Damn it! Ho can it be? These should have triggered the NameError...
Thanks for pointing that out.

Comment (das-g) [reply tests/test_aliases.py:135]: Maybe its different within an alias than when the command line is issued explicitly? Just guessing here, I don't yet know the innards of xonsh.

Comment (laloch) [reply tests/test_aliases.py]: Should be OK now. Thanks for the review.

**Comment by laloch** [comment]
The `p_factor_unary` is not really that important. It's just a part of the Python grammar constructing `UnaryOp` operator AST node. The real evil was happening in https://github.com/xonsh/xonsh/blob/14771a7e51441144148399ab286c81c0d5edbb3d/xonsh/ast.py#L396
When our `CtxAwareTransformer` decides, that the current input cannot be successfully evaluated as a Python code in the current scope, it calls `try_subproc_toks` that tries to find some sense in the input and to replace it with `![subproc args]` uncaptured subprocess construct. It begins by finding the first and the last character index (column) of the current part of the syntax tree within the current input line. The `maxcol` value, that should be the position of the last input character of the current AST node, is derived from the highest value in all of `col_offset` attributes of the current syntax tree. The issue you discovered was caused by a bug in `UnaryOp`'s `col_offset` value, that was pointing behind the text of its argument. It confused `try_subproc_toks` into thinking that the input was not meant to be a subprocess invocation and the whole story ended in a futile attempt to evaluate `(foo) - (bar) - (-version)` as Python code, which obviously led to `NameError`, since there was no `foo` symbol defined.

**Comment by gforsyth** [comment]
And thanks @das-g for looking it over and testing it out!

**Pull request merged**